src/john.c

```diff
@@ -1721,7 +1729,9 @@ static void john_run(void)
 				                           &database);
 			else
 				test_db = &database;
+			self_test_running = 1;
 			where = fmt_self_test(database.format, test_db);
+			self_test_running = 0;
```

Comment:
We could possibly have this variable in `formats.[ch]` and set/reset it from within `fmt_self_test()`. That would cover other possible calls to `fmt_self_test()` from elsewhere. But I'm OK with the current way of doing it as well.